### PR TITLE
Ft verify email

### PR DESCRIPTION
**Pull request by iamcristos**
# Description
This task creates a function that sends email to user when they signup.
This feature also contains '/confirm' endpoint that confirms a User signup
Add attribution link to Readme and code

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change status

- [ ] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

Please describe steps taken to ensure this feature is functional and does not break other features:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation, if relevant
- [ ] I have run the app using my feature and ensured that no functionality is broken
- [ ] My changes generate no new warnings
- [ ] There are no merge conflicts

### CODE REVIEW

File: api/controllers/user.js [hunk 1]
```diff
@@ -1,21 +1,47 @@
...
+        }
         const token = await generateToken(user);
+        await sendEmailConfirmAccount(user, token,'frontend url')
         return successResponse(res, 201, {msg: 'Usercreated', token})
       }
       return errorHelper(res,400, {
         error: "Could not create Profile"
       });
     } catch (error) {
+      console.error(error)
```
Comment (OloruntobiAwoderu) [inline api/controllers/user.js]: I think you should be returning the error and not console. error

File: api/helpers/jwt.js [hunk 2]
```diff
@@ -17,6 +18,18 @@ async function generateToken(user) {
...
 }
+
+async function decodeToken(token) {
+  try {
+    const decoded = jwt.verify(token, JWTSecret)
+    const user = await User.findById(decoded.subject).exec();
+    if(!user) throw Error('no such user');
+    return user;
+  } catch (error) {
+    return null
```
Comment (OloruntobiAwoderu) [inline api/helpers/jwt.js]: Wouldn't it be better to return the error message

File: api/controllers/user.js (hunk 1 above)
Comment (iamcristos) [reply api/controllers/user.js]: yea thanks

File: api/helpers/jwt.js (hunk 2 above)
Comment (iamcristos) [reply api/helpers/jwt.js]: If we resolve the error here when using this helper method somewhere else we will not get to the catch block

Comment (OloruntobiAwoderu) [reply api/helpers/jwt.js]: Okay

**Pull request merged**